spec/projection_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'check_graphite/projection'
+require 'nagios_check'
+
+describe describe CheckGraphite::Projection do
+  let :check do
+    Class.new do
+      attr_reader :values
+      include NagiosCheck
+      include CheckGraphite::Projection
+    end.new
+  end
+
+  describe 'primary value' do
+    subject do
+      check.prepare
+      check.send(:parse_options, ['--projection', projection])
+      check.options.processor.call(datapoints)
+      check.values.first[1]
+    end
+
+    context 'given a linearly decreasing series and a projection of 2 sec' do
+      let(:projection) { '2sec' }
+      let(:datapoints) { [[10,0], [9,1], [8,2]] }
+
+      it { should be_within(0.01).of(6) }
```

Comment:
I'm not convinced by these description-less tests, I don't like the test descriptions. This one would become "should be within 0.01 of 6" for example. That doesn't tell me anything. It doesn't say why that is good or why it's right. Something like 'predicts the next value using a linear model' would better explain what's going on.